### PR TITLE
LMBQ-276: Improving WithEditor

### PR DESCRIPTION
**Pull request by Psichorex**
[LMBQ-276](https://lombiq.atlassian.net/browse/LMBQ-276)

[LMBQ-276]: https://lombiq.atlassian.net/browse/LMBQ-276?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Libraries/Lombiq.HelpfulLibraries
Comment (MZole) [inline src/Libraries/Lombiq.HelpfulLibraries:1]: There are a couple of places where this new Enum could be used already, please use it where we can, so others can find this easier and start using this as well.

**Review by MZole — CHANGES_REQUESTED** [review]

File: src/Libraries/Lombiq.HelpfulLibraries
Comment (Psichorex) [reply src/Libraries/Lombiq.HelpfulLibraries:1]: Addressed.

File: src/Modules/Lombiq.TrainingDemo
Comment (MZole) [inline src/Modules/Lombiq.TrainingDemo:1]: Open a PR for this, please.

File: src/Libraries/Lombiq.HelpfulLibraries
Comment (MZole) [reply src/Libraries/Lombiq.HelpfulLibraries:1]: There should be also an example in HE `SiteTextMigrations`.

**Review by MZole — CHANGES_REQUESTED** [review]

**Review by MZole — APPROVED** [review]

**Review by MZole — APPROVED** [review]

**Pull request merged**